### PR TITLE
feat: decode logs from TXs submitted outside `BaseInvocationScope`

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/1941

### CODE REVIEW

**Comment by Torres-ssf** [comment]
Blocked by: https://github.com/FuelLabs/fuels-ts/pull/1936

I will rebase this after https://github.com/FuelLabs/fuels-ts/pull/1936 got merged

**Review by arboleya — APPROVED** [review]

File: packages/fuel-gauge/src/advanced-logging.test.ts [hunk 1]
```diff
@@ -160,33 +148,135 @@ describe('Advanced Logging', () => {
...
+
+      await wallet.fund(request, requiredQuantities, maxFee);
+
+      const tx = await wallet.sendTransaction(request);
+
+      const { logs } = await tx.waitForResult();
+
+      expect(logs).toBeDefined();
+
+      expect(expecedLogs).toStrictEqual(logs);
```
Comment (petertonysmith94) [inline packages/fuel-gauge/src/advanced-logging.test.ts:279]:
```suggestion
      expect(logs).toStrictEqual(expecedLogs);
```

File: packages/fuel-gauge/src/advanced-logging.test.ts [hunk 2]
```diff
@@ -160,33 +148,135 @@ describe('Advanced Logging', () => {
...
+    });
+
+    it('when using InvocationScope', async () => {
+      const script = new Script(binHexlified, abiContents, wallet);
+      const { logs } = await script.functions
+        .main(advancedLogId, otherLogId, amount)
+        .addContracts([advancedLogContract, otherAdvancedLogContract])
+        .call();
+
+      expect(expecedLogs).toStrictEqual(logs);
```
Comment (petertonysmith94) [inline packages/fuel-gauge/src/advanced-logging.test.ts:249]:
```suggestion
      expect(logs).toStrictEqual(expecedLogs);
```

**Review by petertonysmith94 — APPROVED** [review]

File: packages/fuel-gauge/src/advanced-logging.test.ts (hunk 2 above)
Comment (nedsalk) [reply packages/fuel-gauge/src/advanced-logging.test.ts:249]: There's a typo `expecedLogs` -> `expectedLogs`

Comment (nedsalk) [reply packages/fuel-gauge/src/advanced-logging.test.ts:249]: Save us from airdrop farmers and fix it :sweat_smile:

Comment (Torres-ssf) [reply packages/fuel-gauge/src/advanced-logging.test.ts:249]: @nedsalk you make a strong case 🤣

**Review by arboleya — APPROVED** [review]

**Review by petertonysmith94 — APPROVED** [review]

**Pull request merged**